### PR TITLE
fuzz: bazelify fuzzit runs and run fuzzit in RBE

### PR DESCRIPTION
**Pull request by lizan**
Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:

### CODE REVIEW

**Comment by lizan** [comment]
cc @yevgenypats

**Comment by lizan** [comment]
cc @htuch I suspect this is also one of the cost factor of RBE egress, because ASAN fuzzer binaries are big.

**Comment by htuch** [comment]
Awesome. If we can get @asraa and @yevgenypats to give this a review and sign-off, I can do the final pass.

File: bazel/fuzzit_wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+set -e
+
+# run fuzzing regression or upload to Fuzzit for long running fuzzing job depending on whether FUZZIT_API_KEY is set
+
+FUZZIT="${TEST_SRCDIR}/fuzzit_linux/fuzzit"
+
+FUZZER_BINARY=$1
+FUZZIT_TARGET_NAME="$(basename $1 | sed -e s/_fuzz_test_with_libfuzzer$// -e s/_/-/g)"
+
+if [[ ! -z "${FUZZIT_API_KEY}" ]]; then
```
Comment (yevgenypats) [inline bazel/fuzzit_wrapper.sh]: I think the previous condition was this one https://github.com/envoyproxy/envoy/pull/8835/files#diff-abef39f4a7cbfbedd8850dcd8fbcc792L145

as we wanted the fuzzing to run on push to master and not on pull-requests. This is becuase fuzzing is essentially a long running job that only stops when a new crash is found or when new code is pushed and then the new fuzzers starts and the old one stops (keeping the corpus between runs).

File: ci/run_fuzzit.sh [hunk 2]
```diff
@@ -1,51 +0,0 @@
...
-  fi
-done
-
-
-# run fuzzing regression or upload to Fuzzit for long running fuzzing job ($1 is either local-regression or fuzzing)
-wget -O fuzzit https://github.com/fuzzitdev/fuzzit/releases/latest/download/fuzzit_Linux_x86_64
-chmod a+x fuzzit
-
-PREFIX=$(realpath /build/tmp/_bazel_bazel/*/execroot/envoy/bazel-out/k8-fastbuild/bin)
-SLOW_TARGETS=("access-log-formatter" "h1-capture" "h1-capture-direct-response" "response-header" "request-header")
```
Comment (yevgenypats) [inline ci/run_fuzzit.sh]: Is this left out intentionally in the new commit?

**Comment by yevgenypats** [comment]
@htuch @lizan looks good, had a few comments/questions. I'm not very familiar with RBE but where can I see a test run of the PR? is it this one https://dev.azure.com/cncf/envoy/_build/results?buildId=17550 or we can see it only after merge.

File: bazel/fuzzit_wrapper.sh (hunk 1 above)
Comment (lizan) [reply bazel/fuzzit_wrapper.sh]: @yevgenypats yes this is intended, I merged two jobs into one. In our CI setup, PR runs doesn’t have FUZZIT_API_KEY set so on master this run fuzzing + local-regression, and on PR run this only run local-regression.

File: ci/run_fuzzit.sh (hunk 2 above)
Comment (lizan) [reply ci/run_fuzzit.sh]: intentionally, I set the timeout to 1200s (see do_ci.sh) for each fuzz and so far no fuzz timed out.

**Comment by lizan** [comment]
https://dev.azure.com/cncf/envoy/_build/results?buildId=17550 only has local regression, I tested fuzzing locally too.

**Comment by yevgenypats** [comment]
lgtm

**Comment by lizan** [comment]
@htuch can you take a final look?

**Pull request merged**